### PR TITLE
Refactor groundparticles gamma

### PR DESCRIPTION
**Pull request by tomkooij**
Refactored code for detector response to gamma's

Refactored code todo:
- [x] delete old branch groundparticles_gamma
- [x] added separate class GroundParcticleSimulationGamma() 
- [x] compare GroundParticlesSimulation() and GroundParticlesGammaSimulation()
- [x] check digitisation algorithm and implementation
    - [x] check detection probability for compton scattering
    - [x] check detection probability for pair production
    - [x] check energy transfer for compton scattering
    - [x] check energy transfer for pair production
- [x] add unittests
- [x] PEP8

- [x] remove numpy.compress from GroundParticlesGammaSimulation.get_particles_in_detector() https://github.com/HiSPARC/sapphire/blob/refactor_groundparticles_gamma/sapphire/simulations/groundparticles.py#L322
- [x] document  `_pair_mean_free_path(Energy)`: from XCOM2 (_not the game_)

- [x] Acceptance test for GroundParticlesGammaSimulation

- [x] The test for `simulate_detector_mips_gammas` needs to be updated to have multiple non-zero theta and multiple p.
- [x] Also the test data for the gamma simulation needs to be updated.
- [x] And perhaps a test for a very inclined shower causing the scintillator depth to reach the maximum..

### CODE REVIEW

**Comment by 153957** [comment]
- added PEP8 to todo list (mostly missing spaces around operators).

**Comment by tomkooij** [comment]
Argh! Fixed in 4103128.
Not all PEP8 checker are consistent. Travis is still complaining about some minor things, but my code passes http://pep8online.com/

**Comment by 153957** [comment]
We use flake8 for testing, which also complains about non-PEP8 things. flake8 is a combination of pep8 and pyflakes, pyflakes checks code consistency: unused imports, missing variables, etc..
Some of the errors reported by PEP8 are intentionally wrong, like the matrix in `geographic.py`, for better readability. This is why we do not let the flake8 test cause a fail.

**Comment by tomkooij** [comment]
I missed the spaces around `+` in line 271: fixed in 96addb3

flake8 complains about missing whitespaces in line 441: return -0.806 * (E**2) + 8.46 * E + 6.36 
Seems to want a space after `-` , which is ridiculous and decreases readability. Ignored.

**Comment by 153957** [comment]
no, it wants spaces around `**`

**Comment by tomkooij** [comment]
Sigh. You're probably right. Fixed in 1fd364b 
This PEP8 thing really destroys ```git blame``` functionality. (I'd better flake8 *before* commiting a new feature)

**Comment by tomkooij** [comment]
The energy transfer from compton scattering is a "magic" table right now. I (still) need to replace that with the algorithm that created the table.

That is (only slightly) slower but easier to maintain and much easier to debug/verify.

File: sapphire/simulations/groundparticles.py [hunk 1]
```diff
@@ -230,14 +228,364 @@ def get_particles_in_detector(self, detector):
...
+                # based on remaining scinitilator depth
+                maximum_energy_deposit_in_MIPS = \
+                    ((scintilator_depth - depth_pair) / scintilator_depth) * \
+                    max_E / MIP / costheta
+
+                # 1.022 MeV used for creation of two particles
+                # all the rest is electron kinetic energy
+                energy_deposit_in_MIPS = (energy - 1.022) / MIP
+
+                mips += np.minimum(maximum_energy_deposit_in_MIPS,
```
Comment (tomkooij) [inline sapphire/simulations/groundparticles.py]: `np.minimum` is element-wise minimum and thus returns a vector. 

But this is supposed to be "per item", not vectorised!

mips should be a scalar, not a vector! This needs fixing.

**Comment by 153957** [comment]
Remember to also add the new gammas module to __init__.py and docs.

**Comment by tomkooij** [comment]
No clue why the Travis fails on the PR but passes the other one *and* `make test` passed.

Perhaps so "order of the tests" issue with `@patch`. 

Anyway, this only needs an acceptance test and should be finished.

**Comment by tomkooij** [comment]
This is an interesting read: http://www.voidspace.org.uk/python/mock/patch.html#where-to-patch

**Comment by davidfokkema** [comment]
It is. I must've read it about a dozen times (and counting).

**Comment by 153957** [comment]
Ready to merge?

**Comment by tomkooij** [comment]
@153957, yes, ready to merge. Feel free to merge this, if it looks okay to you.

**Comment by 153957** [comment]
Almost, I have found a potential bug (and some code style/docstrings that may need to be improved).

**Comment by 153957** [comment]
The test for `simulate_detector_mips_gammas` needs to be updated to have multiple non-zero theta and multiple p.

**Pull request merged**